src/ray/gcs/gcs_server/gcs_job_manager.cc

```diff
@@ -187,13 +206,37 @@ void GcsJobManager::HandleGetAllJobInfo(rpc::GetAllJobInfoRequest request,
         job_api_data_keys.push_back(job_data_key);
         job_data_key_to_indices[job_data_key].push_back(i);
       }
+
+      // Get is_running_tasks from the core worker for the driver.
+      auto client = core_worker_clients_.GetOrConnect(data.second.driver_address());
+      std::unique_ptr<rpc::NumPendingTasksRequest> request(
+          new rpc::NumPendingTasksRequest());
+      client->NumPendingTasks(
+          std::move(request),
+          [reply, i, num_processed_jobs, try_send_reply](
+              const Status &status,
+              const rpc::NumPendingTasksReply &num_pending_tasks_reply) {
+            if (!status.ok()) {
+              RAY_LOG(ERROR) << "Failed to get is_running_tasks from core worker: "
+                             << status.ToString();
+            }
```

Comment:
Not sure the best way to handle the error here, or in what situations this can happen.  I think maybe this can happen if the core worker hasn't started yet.

If we default to is_running_tasks=False in this case, then cluster managers might prematurely shut down the cluster.  Buf it we default to is_running_tasks=True, the cluster might stay alive forever.

Reply:
Another option is to expose the entire `int num_pending_tasks` instead of just the `bool is_running_tasks`, and set `num_pending_tasks` to -1 if we can't connect to the core worker.